pkg/proc/native/proc_linux.go

```diff
@@ -689,6 +706,46 @@ func (dbp *nativeProcess) EntryPoint() (uint64, error) {
 	return linutil.EntryPointFromAuxv(auxvbuf, dbp.bi.Arch.PtrSize()), nil
 }
 
+func (dbp *nativeProcess) SupportsBPF() bool {
+	return true
+}
+
+func (dbp *nativeProcess) SetUProbe(fnName string, args []ebpf.UProbeArgMap) {
+	// Lazily load and initialize the BPF program upon request to set a uprobe.
+	if dbp.os.ctx == nil {
+		dbp.os.ctx, _ = ebpf.LoadEBPFTracingProgram()
+	}
+
+	// We only allow up to 6 args for a BPF probe.
+	// Return early if we have more.
+	if len(args) > 6 {
+		return
+	}
+
+	debugname := dbp.bi.Images[0].Path
+	offset, err := ebpf.SymbolToOffset(debugname, fnName)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
```

Comment:
I think it would be better to return the error rather than calling os.Exit.

Reply:
Done.